resources/assets/js/annotations/components/annotationCanvas/polygonBrushInteraction.vue

```diff
@@ -100,33 +100,44 @@ export default {
             }
         },
     },
-    created() {
-        if (this.canAdd) {
-            Keyboard.on('e', this.togglePolygonBrush, 0, this.listenerSet);
-        }
-
-        if (this.canModify) {
-            Keyboard.on('r', this.togglePolygonEraser, 0, this.listenerSet);
-            Keyboard.on('t', this.togglePolygonFill, 0, this.listenerSet);
-        }
-
-        if (this.canAdd || this.canModify) {
-            this.$watch('interactionMode', this.toggleCurrentInteraction);
+    watch: {
+        interactionMode(mode) {
+            if (this.canAdd || this.canModify) {
+                this.toggleCurrentInteraction(mode);
+            }
+        },
+        canAdd(state) {
+            if (state) {
+                Keyboard.on('e', this.togglePolygonBrush, 0, this.listenerSet);
+            }
+            else {
+                Keyboard.off('e', this.togglePolygonBrush, 0, this.listenerSet);
+            }
+        },
+        canModify(state) {
+            if (state) {
+                Keyboard.on('r', this.togglePolygonEraser, 0, this.listenerSet);
+                Keyboard.on('t', this.togglePolygonFill, 0, this.listenerSet);
+            }
+            else {
+                shiftClickSelectInteraction.setActive(false);
+                Keyboard.off('r', this.togglePolygonEraser, 0, this.listenerSet);
+                Keyboard.off('t', this.togglePolygonFill, 0, this.listenerSet);
+            }
         }
     },
     mounted() {
-        if (this.canModify) {
-            shiftClickSelectInteraction = new SelectInteraction({
-                condition(e) {
-                    return click(e) && shiftKeyOnly(e);
-                },
-                style: Styles.highlight,
-                layers: [this.annotationLayer],
-                features: this.selectInteraction.getFeatures(),
-                multi: true,
-            });
-            shiftClickSelectInteraction.on('select', this.handleFeatureSelect);
-        }
+        shiftClickSelectInteraction = new SelectInteraction({
+            condition(e) {
+                return click(e) && shiftKeyOnly(e);
+            },
+            style: Styles.highlight,
+            layers: [this.annotationLayer],
+            features: this.selectInteraction.getFeatures(),
+            multi: true,
+        });
+        shiftClickSelectInteraction.on('select', this.handleFeatureSelect);
+        shiftClickSelectInteraction.setActive(false);
```

Comment:
This is not required because the interaction is not added to the map initially. However, the current implementation here is bad because once the `shiftClickSelectInteraction` is added to the map (in `toggleCurrentInteraction()`) it is never removed again. Instead, it should be created and removed just like the `ModifyPolygonBrushInteraction` in `toggleCurrentInteraction`. Maybe you can implement this change right in this PR, too.

Reply:
That kind of implementation also occurs in [translateInteractions](https://github.com/biigle/core/blob/fdde8273c2f2f3dbc5b4cafb2c368219d960fd82/resources/assets/js/annotations/components/annotationCanvas/translateInteraction.vue#L38-L41) and in [magicWandInteraction](https://github.com/biigle/core/blob/fdde8273c2f2f3dbc5b4cafb2c368219d960fd82/resources/assets/js/annotations/components/annotationCanvas/magicWandInteraction.vue#L75-L82). Should the interactions be deleted there too?

Reply:
No, the places you have linked add the interactions once to the map and then work with `setActive()` to enable and disable the interactions. This is fine and different to the polygon brush I was talking about. It uses `addInteraction()` and `removeInteraction()`. The `shiftClickSelectInteraction` is only ever added and never removed, which is the problem here.

However, looking at it again, the `shiftClickSelectInteraction` is disabled initially (`setActive(false)`) and never enabled in the file. This must be fixed. You can then work with `setActive()` instead of `add`/`removeInteraction()` here, too. So you can add `shiftClickSelectInteraction` to the map in `mounted()` but disable it and then conditionally disable/enable it in `toggleCurrentInteraction()`.